### ISSUE TITLE
@typescript-eslint/restrict-plus-operands: alias types aren't being checked properly

### ISSUE THREAD

**Issue by jpike88** [typescript-eslint/typescript-eslint#2332]
**Repro**

Two types are string, but are declared via an alias.

See screenshots.

<img width="922" alt="Screen Shot 2020-07-28 at 2 43 59 PM" src="https://user-images.githubusercontent.com/9585787/88628602-c84d9200-d0e0-11ea-996b-fddfdd597c7e.png">
<img width="913" alt="Screen Shot 2020-07-28 at 2 45 19 PM" src="https://user-images.githubusercontent.com/9585787/88628743-f8953080-d0e0-11ea-9da9-a3a6e564fd65.png">

Both types are strings. But they are still false positives via the restrict-plus-operands rule. Maybe it's because of aliasing?

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.0` |
| `@typescript-eslint/parser`        | `3.7.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `12.8.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Aliases shouldn't do anything - typescript will resolve the aliases and tell us the underlying type.

Generally when something weird like this it generally means that something is misconfigured, causing  typescript to report the wrong types to the linter.

If you remove the `+ ' ' +` from the expression and just do `f.firstName + f.lastName`, does it still error?
Just hazarding a guess - I suspect that this will report the message `Operands of '+' operation must either be both strings or both numbers.` (different to the current message which also includes the ending `Consider using a template literal`). If it does, then it's definitely a config issue as the types are likely resolving to `any`.

If not, then could you please provide a minimal repro for this? A gist or a repro repo would be great to help investigating this.

**Comment by jpike88** [comment]
Removing the ```' '``` doesn't make the error go away, correct.

**Comment by bradzacher** [comment]
If you dump a bit more info about your config (tsconfig and eslintrc) and code, I might be able to help you investigate potential issues.

I'm assuming that the `User` type comes from another package (npm or local).

**Comment by jpike88** [comment]
The User type comes from a project aliased as ```@shared```, as you can see in tsconfig below.

tsconfig:

```
{
	"compileOnSave": true,
	"angularCompilerOptions": {
		"enableIvy": false,
		"fullTemplateTypeCheck": false,
		"disableTypeScriptVersionCheck": true
	},
	"compilerOptions": {
		"plugins": [
			{
				"name": "@angular/language-service"
			}
		],
		"moduleResolution": "node",
		"target": "ESNext",
		"lib": ["esnext", "dom"],
		"emitDecoratorMetadata": true,
		"outDir": "../dist",
		"sourceMap": true,
		"allowSyntheticDefaultImports": true,
		"noEmitHelpers": false,
		"noUnusedLocals": true,
		"noUnusedParameters": true,
		"forceConsistentCasingInFileNames": true,
		"noImplicitThis": true,
		"skipLibCheck": true,
		"experimentalDecorators": true,
		"baseUrl": "./",
		"paths": {
			"@client/*": ["./components/*"],
			"@shared/*": ["./../../shared/*"]
		}
	},
	"exclude": ["./../node_modules"]
}
```


.eslint.rc
```
module.exports = {
	env: {
		browser: true,
		es6: true,
	},
	extends: [
		'plugin:@typescript-eslint/recommended',
		'plugin:@typescript-eslint/recommended-requiring-type-checking',
	],
	parser: '@typescript-eslint/parser',
	plugins: [
		'@typescript-eslint',
		'ban',
		'import',
		'jsdoc',
		// 'prettier',
	],
	rules: {
		'@typescript-eslint/adjacent-overload-signatures': 'error',
		'@typescript-eslint/array-type': [
			'error',
			{
				default: 'array-simple',
			},
		],
		'@typescript-eslint/camelcase': 'off',
		'@typescript-eslint/await-thenable': 'error',
		'@typescript-eslint/ban-types': 'off',
		'@typescript-eslint/no-unsafe-assignment': 'off',
		'@typescript-eslint/explicit-module-boundary-types': 'off',
		'@typescript-eslint/no-unsafe-call': 'off',
		'@typescript-eslint/ban-ts-comment': 'off',
		'@typescript-eslint/no-unsafe-member-access': 'off',
		'@typescript-eslint/no-unsafe-return': 'off',
		'@typescript-eslint/naming-convention': [
			'error',
			{
				selector: 'class',
				format: ['PascalCase'],
			},
		],
		'@typescript-eslint/switch-exhaustiveness-check': 'error',
		'@typescript-eslint/consistent-type-assertions': 'error',
		indent: 'off',
		'@typescript-eslint/indent': [
			'error',
			'tab',
			{ FunctionDeclaration: { body: 2 } },
		],
		'@typescript-eslint/explicit-function-return-type': 'off',
		'@typescript-eslint/interface-name-prefix': 'off',
		'@typescript-eslint/unbound-method': 'off',
		'@typescript-eslint/member-delimiter-style': [
			'error',
			{
				multiline: {
					delimiter: 'semi',
					requireLast: true,
				},
				singleline: {
					delimiter: 'semi',
					requireLast: false,
				},
			},
		],
		'@typescript-eslint/member-ordering': 'off',
		'@typescript-eslint/no-empty-function': 'error',
		'@typescript-eslint/no-empty-interface': 'error',
		'@typescript-eslint/prefer-regexp-exec': 'off',
		'@typescript-eslint/no-explicit-any': 'off',
		'@typescript-eslint/no-misused-promises': [
			'error',
			{
				checksVoidReturn: false,
			},
		],
		'@typescript-eslint/no-misused-new': 'error',
		'@typescript-eslint/no-namespace': 'off',
		'@typescript-eslint/no-parameter-properties': 'off',
		'@typescript-eslint/no-use-before-define': 'off',
		'@typescript-eslint/no-var-requires': 'error',
		'@typescript-eslint/prefer-for-of': 'error',
		'@typescript-eslint/prefer-function-type': 'error',
		'@typescript-eslint/prefer-namespace-keyword': 'error',
		'ban/ban': [
			2,
			{
				name: ['*', 'forEach'],
				message: "Don't use foreach on an array, use forof",
			},
		],
		'@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
		'@typescript-eslint/semi': ['off'],
		'@typescript-eslint/triple-slash-reference': [
			'error',
			{
				path: 'always',
				types: 'prefer-import',
				lib: 'always',
			},
		],
		'@typescript-eslint/unified-signatures': 'error',
		'arrow-parens': ['off', 'always'],
		camelcase: 'off',
		'comma-dangle': 'off',
		complexity: 'off',
		'constructor-super': 'error',
		'dot-notation': 'error',
		eqeqeq: ['error', 'smart'],
		'guard-for-in': 'error',
		'id-blacklist': [
			'error',
			'any',
			'Number',
			'number',
			'String',
			'string',
			'Boolean',
			'boolean',
		],
		'no-underscore-dangle': ['error', { allow: ['_id'] }],
		'id-match': 'error',
		'import/no-deprecated': 'error',
		'import/order': 'off',
		'jsdoc/check-alignment': 'error',
		'jsdoc/check-indentation': 'error',
		'prefer-arrow-callback': 'error',
		'jsdoc/newline-after-description': 'error',
		'max-classes-per-file': 'off',
		'max-len': [
			'error',
			{
				ignorePattern: '\t',
				tabWidth: 1,
				ignoreStrings: true,
				code: 120,
			},
		],
		'new-parens': 'error',
		'no-bitwise': 'error',
		'no-caller': 'error',
		'prefer-spread': 'off',
		'@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '_' }],
		'no-cond-assign': 'error',
		'no-console': [
			'error',
			{
				allow: [
					'warn',
					'dir',
					'timeLog',
					'assert',
					'clear',
					'count',
					'countReset',
					'group',
					'groupEnd',
					'table',
					'info',
					'dirxml',
					'error',
					'groupCollapsed',
					'Console',
					'profile',
					'profileEnd',
					'timeStamp',
					'context',
				],
			},
		],
		'no-debugger': 'error',
		'no-empty': 'error',
		'no-eval': 'error',
		'no-fallthrough': 'off',
		'no-invalid-this': 'off',
		'no-multiple-empty-lines': 'error',
		'no-new-wrappers': 'error',
		'@typescript-eslint/indent': 'off',
		'no-restricted-imports': [
			'error',
			{
				paths: [
					'@angular/compiler/src/core',
					'@ngx-translate/core/public_api',
					{
						name: 'ag-grid-community',
						importNames: ['Environment'],
					},
				],
			},
		],
		'no-restricted-syntax': ['error', 'ForInStatement'],
		'no-return-await': 'error',
		'no-shadow': [
			'off',
			{
				hoist: 'all',
			},
		],
		'no-throw-literal': 'off',
		'no-trailing-spaces': 'error',
		'no-undef-init': 'error',
		'no-underscore-dangle': 'off',
		'no-unsafe-finally': 'error',
		'no-unused-expressions': 'error',
		'no-unused-labels': 'error',
		'no-var': 'error',
		'object-shorthand': 'error',
		'one-var': ['error', 'never'],
		'prefer-const': 'error',
		// 'prettier/prettier': 'error',
		'quote-props': ['error', 'as-needed'],
		radix: 'error',
		'space-before-function-paren': [
			'error',
			{
				anonymous: 'always',
				named: 'never',
				asyncArrow: 'always',
			},
		],
		'spaced-comment': [
			'error',
			'always',
			{
				markers: ['/'],
			},
		],
		'use-isnan': 'error',
		'valid-typeof': 'off',
	},
};
```

**Comment by bradzacher** [comment]
The eslintrc doesn't contain a `parserOptions` - how are you configuring the tsconfigs that are passed to the linter?

**Comment by jpike88** [comment]
I forgot to include this .eslintrc.js that is in my client folder

```
module.exports = {
	extends: ['../.eslintrc.js'],
	parserOptions: {
		project: 'client/src/tsconfig.json',
		sourceType: 'module',
	},
	rules: {
		'@typescript-eslint/no-inferrable-types': 0,
	},
};
```

**Comment by jpike88** [comment]
Also, this issue is only occurring in VSCode.

I have a project structure like this:

```
root /
        / vscodeworkspace.code-workspace
        / .eslintrc.js     <-- contains all the rules
        / - subproject
                 / - .eslintrc.js  
                 / - tsconfig.json
        / - shared typescript files (where the User type is sitting)
        / - subproject2
```

eslint works fine for everything else, just this rule is being an issue.
